src/prop.ts

```diff
@@ -52,241 +45,255 @@ function baseProp(
 
   const initname = utils.createUniqueID(target);
 
-  decoratorCache.get(initname).decorators.set(key, () => {
-    if (utils.isNotDefined(Type)) {
-      if (Type !== target) { // prevent "infinite" buildSchema loop / Maximum Class size exceeded
-        buildSchema(Type, { _id: typeof rawOptions._id === 'boolean' ? rawOptions._id : true });
-      }
-    }
-    const name: string = utils.getName(target.constructor);
-    rawOptions = Object.assign(rawOptions, {});
-
-    if (!virtuals.get(name)) {
-      virtuals.set(name, new Map());
-    }
-
-    if (utils.isWithVirtualPOP(rawOptions)) {
-      if (!utils.includesAllVirtualPOP(rawOptions)) {
-        throw new NotAllVPOPElementsError(name, key);
-      }
-      virtuals.get(name).set(key, rawOptions);
+  const existingMapForTarget = Reflect.getOwnMetadata(DecoratorKeys.PropProps, target) as DecoratedPropertyMetadataMap;
+  if (!existingMapForTarget) {
```

Comment:
i would prefer if you would use `isNullOrUndefined`

Reply:
good catch.  will do :)